src/Symfony/Component/Messenger/WorkerMetadata.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger;
+
+/**
+ * @author Oleg Krasavin <okwinza@gmail.com>
+ */
+final class WorkerMetadata
+{
+    private $metadata;
+
+    public function __construct(array $metadata)
+    {
+        $this->metadata = $metadata;
+    }
+
+    public function set(array $newMetadata): void
+    {
+        $this->metadata = array_merge($this->metadata, $newMetadata);
+    }
+
+    /**
+     * Returns the queue names the worker consumes from, if "--queues" option was used.
+     * Returns false otherwise.
+     *
+     * @return array|false
+     */
+    public function getQueueNames()
+    {
+        return $this->metadata['queueNames'] ?? false;
```

Comment:
```suggestion
     */
    public function getQueueNames(): ?array
    {
        return $this->metadata['queueNames'] ?? null;
```

this version feels more natural to me

Reply:
I totally agree with you here. The reason why I went with `false` is to keep things consistent with this: https://github.com/symfony/symfony/blob/9260aa6607992503b8063a690692f9ec545b1a07/src/Symfony/Component/Messenger/Worker.php#L73

But I guess we could just change the both cases to `?? null` instead. :)